sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -185,6 +185,13 @@ case class FileSourceScanExec(
     ret
   }
 
+  @transient private lazy val selectedPartitionsCount =
+    if (partitionFilters.exists(ExecSubqueryExpression.hasSubquery)) {
+      None
```

Comment:
why do we do this?

Reply:
This is invoked from `FileSourceScanExec.metadata` and part of `FileSourceScanExec`'s `toString` value too, so it would throw 
```
requirement failed: Subquery subquery250 has not finished
java.lang.IllegalArgumentException: requirement failed: Subquery subquery250 has not finished
	at scala.Predef$.require(Predef.scala:281)
	at org.apache.spark.sql.execution.ScalarSubquery.eval(subquery.scala:95)
```
exception when printing executed plan.

Reply:
is this problem present before https://issues.apache.org/jira/browse/SPARK-25482 ? i.e. Spark 2.4 and prior

Reply:
No. This is required due to this PR. `partitionFilters` always filtered out subqueries. [SPARK-25482](https://issues.apache.org/jira/browse/SPARK-25482) just filtered out subqueries from data filters (pushed down filters) because they were not used but caused double execution.